packages/wonder-blocks-core/components/unique-id-provider.test.js

```diff
@@ -124,4 +125,28 @@ describe("UniqueIDProvider", () => {
             expect(children.mock.calls[2][0]).toBe(children.mock.calls[1][0]);
         });
     });
+
+    describe("inside a NoSSR", () => {
+        test("it should pass an id to its children", () => {
+            // Arrange
+            const foo = jest.fn(() => null);
+            const nodes = (
+                <NoSSR>
+                    {() => (
+                        <UniqueIDProvider>
+                            {(ids) => foo(ids.get(""))}
+                        </UniqueIDProvider>
+                    )}
+                </NoSSR>
+            );
+            const wrapper = mount(nodes);
+
+            // Act
+            wrapper.instance().forceUpdate();
+
+            // Assert
+            expect(foo).toHaveBeenCalled();
+            expect(foo.mock.calls[0][0]).toBeTruthy();
+        });
+    });
```

Comment:
Great test coverage! Thank you for maintaining the "Arrange/Act/Assert" pattern.

Reply:
Thanks for starting the pattern even if I was resistant at first.